modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/queue/KafkaEncoder.scala

```diff
@@ -0,0 +1,53 @@
+package ch.epfl.bluebrain.nexus.iam.service.queue
+
+import ch.epfl.bluebrain.nexus.iam.core.acls._
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.auto._
+import io.circe.syntax._
+import io.circe.{Encoder, Json, Printer}
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.commons.iam.acls.AccessControlList
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Identity
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Identity.{GroupRef, UserRef}
+import io.circe.generic.extras.semiauto.deriveEncoder
+
+// $COVERAGE-OFF$
+trait KafkaEncoder {
```

Comment:
vals are extremely dangerous in traits due to the initialization order, please use defs when depending of yet undefined values.

Why is coverage set to off? the behaviour seems easily testable...